static/js/controllers/contacts.js

```diff
@@ -95,9 +96,17 @@ var SP_ID_MASK = 0xFFFFF8;
               // move it to the top
               contacts.splice(homeIndex, 1);
               contacts.unshift(usersHomePlace);
-            } else if (!$scope.filters.search) {
+            } else if (!$scope.filters.search && !$scope.filters.simprintsIdentities) {
               contacts.unshift(usersHomePlace);
             }
+            if ($scope.filters.simprintsIdentities) {
+              contacts.forEach(function(contact) {
+                var identity = $scope.filters.simprintsIdentities.find(function(identity) {
+                  return identity.id === contact.simprints_id;
```

Comment:
Currently the AndroidApi parses the response as JSON and then passes it on to `Simprints.response` function which handles responses of all types. In retrospect this was probably laziness on my part and we should have a `identifyResponse` and `registerResponse` functions so the response can be handled differently. This would also help for the filtering suggested below.